engine/src/mblhandlers.cpp

```diff
@@ -903,7 +903,11 @@ Exec_stat MCHandleGetPurchaseProperty(void *context, MCParameter *p_parameters)
     
     if (!ctxt.HasError())
     {
-        ctxt.SetTheResultToValue(*t_prop_value);
```

Comment:
You can revert that change, as MCStoreGetPurchaseProperty will never return `nil` with your fix